scripts/behavior.js

```diff
@@ -371,7 +371,129 @@ function executeCommand(command, arguments, cooldown, message, settings, utils,
+                        .then(callback(COMMAND_SUCCESS));
+                }
+
+                break;
+
+            case "subscribe":
+                if (arguments == null){
+                    callback(COMMAND_MISUSE);
+                    break;
+                }
+                let subscribeRoleName = escapeArguments(arguments);
+
+                let specificsSubscribe = utils.getSpecifics(message.guild);
+
+                if(! specificsSubscribe["registeredRoles"].some(role => role === subscribeRoleName)) {
+                    replyToMessage(message, "Role not registered.")
+                        .then(callback(COMMAND_MISUSE));
+                    break;
+                }
+
+                let roleToSubscribe = Array.from(message.guild.roles.values()).find(role => role.name === subscribeRoleName);
+
+                if(roleToSubscribe !== undefined) {
+                    message.guild.fetchMember(message.author)
+                        .then(guildMember => {
+                            guildMember.addRole(roleToSubscribe);
+                            replyToMessage(message, "You have been subscribed to " + roleToSubscribe.name)
+                                .then(callback(COMMAND_SUCCESS));
+                        })
+                } else {
+                    replyToMessage(message, "Role not present on discord server.")
+                        .then(callback(COMMAND_SUCCESS));
+                }
+
+                break;
+
+            case "unsubscribe":
+                if (arguments == null){
+                    callback(COMMAND_MISUSE);
+                    break;
+                }
+                let unsubscribeRoleName = escapeArguments(arguments);
+
+                let specificsUnsubscribe = utils.getSpecifics(message.guild);
+
+                if(! specificsUnsubscribe["registeredRoles"].some(role => role === unsubscribeRoleName)) {
+                    replyToMessage(message, "Role not registered.")
+                        .then(callback(COMMAND_MISUSE));
+                    break;
+                }
+
+                let roleToUnsubscribe = Array.from(message.guild.roles.values()).find(role => role.name === unsubscribeRoleName);
+
+                if(roleToUnsubscribe !== undefined) {
+                    message.guild.fetchMember(message.author)
+                        .then(guildMember => {
+                            guildMember.removeRole(roleToUnsubscribe);
+                            replyToMessage(message, "You have been unsubscribed from " + roleToUnsubscribe.name)
+                                .then(callback(COMMAND_SUCCESS));
+                        })
+                } else {
+                    replyToMessage(message, "Role not present on discord server.")
+                        .then(callback(COMMAND_SUCCESS));
+                }
+
                 break;
 		}

```

Comment:
Please add a !subscriptions (or name it what you like) to list the current test rols (mod only)
and a !subscribers <role> or !showsubscribers <role> to list the subscribers of a certain role and their count (mod only aswell) (name it what you like)

Also, please update the readme :)

Reply:
About show subscribers. I would have to query the discord server for that? Isn't that discord's job?

Reply:
Good question, i'm not sure, checkout discord JS documentation. It may or may not be possible to enumerate discord users per role.

Reply:
It will be possible, question is if it makes sense to do it in this bot